test/functional/feature_assumeutxo.py

```diff
@@ -0,0 +1,223 @@
+#!/usr/bin/env python3
+# Copyright (c) 2021 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test for assumeutxo, a means of quickly bootstrapping a node using
+a serialized version of the UTXO set at a certain height, which corresponds
+to a hash that has been compiled into bitcoind.
+
+The assumeutxo value generated and used here is committed to in
+`CRegTestParams::m_assumeutxo_data` in `src/chainparams.cpp`.
+"""
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import assert_equal, wait_until_helper
+
+START_HEIGHT = 199
+SNAPSHOT_BASE_HEIGHT = 299
+FINAL_HEIGHT = 399
+COMPLETE_IDX = {'synced': True, 'best_block_height': FINAL_HEIGHT}
+
+
+class AssumeutxoTest(BitcoinTestFramework):
```

Comment:
> Interesting test cases could be loading an assumeutxo snapshot file with:

As you probably know, a lot of these cases are already covered by the unittests. I definitely agree it would be nice to test them from within the functional tests, but actually generating such invalid snapshots will require some thought.

I've consolidated all the TODOs in the test's docstring.